File: php/oop-study/lesson3/Abstract/Command.php
<?php
	abstract class Command{
		public abstract function execute();		
		public final function doExecute(){
			//Тут выполняется блок инструкций до выполнения команды
			echo "Command ".get_class($this)." now be executed. Pre actions done!<br>";
			$this->execute();
		}
	}
?>

File: java/bsuStudy/command/Command.java
public abstract class Command{
	public void logAction(){
		String time="current datetime";
		//поток вывода в файл!
		System.out.println(time+": command executed - "+this.getClass());	
	}
	public abstract void doExecute();
	public void execute(){
		logAction();
		doExecute();
	}
}